packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -274,59 +332,60 @@ export class ProjectLifecycleManager {
 
     this._projectRoot = projectRoot
     this._initializedProject = undefined
-    this._pendingMigrationInitialize = pDefer()
-    this.legacyPluginGuard()
-    Promise.resolve(this.ctx.browser.machineBrowsers()).catch(this.onLoadError)
-    this.verifyProjectRoot(projectRoot)
-    const packageManagerUsed = this.getPackageManagerUsed(projectRoot)
 
     this.resetInternalState()
+
+    this._configManager = this.createConfigManager()
+
+    // Preemptively load these so that they are available when we need them later
+    this.ctx.browser.machineBrowsers().catch(this.onLoadError)
+
+    const packageManagerUsed = this.getPackageManagerUsed(projectRoot)
+
     this.ctx.update((s) => {
       s.currentProject = projectRoot
       s.packageManager = packageManagerUsed
     })
 
+    if (this.readyToInitialize(this._projectRoot)) {
+      this._configManager.initializeConfig().catch(this.onLoadError)
```

Comment:
Why is this statement not awaited?
It could be because `setCurrentProject()` needs to be called in the constructor (and the contructor cannot be async).

If yes, I would suggest we split it off in a second function called after creation.